src/components/equations/ItemEffects.ts

```diff
@@ -0,0 +1,22 @@
+import { IMove } from "battlemovr";
+import { GeneralComponent } from "gamestartr";
+
+import { FullScreenPokemon } from "../../FullScreenPokemon";
+import { IPokemon } from "../Battles";
+
+export class ItemEffects<TGameStartr extends FullScreenPokemon> extends GeneralComponent<TGameStartr> {
+
+    /**
+     * Restores a certain amount of PP to a Pokemon's move.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param index   The index of the moveset that is being modified.
+     * @param amount   The amount of PP that is being restored.
+     */
+    public addPP(move: IMove, amount: number) {
+        if (amount < 0) {
+            return;
```

Comment:
This case should basically never happen. Throw an error or don't handle it here (though it'd probably end up getting added in a subsequent unit test)